src/router/validation/subscriptionValidator.test.ts

```diff
@@ -0,0 +1,208 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import { InvalidResourceError, Search } from 'fhir-works-on-aws-interface';
+import ElasticSearchService from '../__mocks__/elasticSearchService';
+import SubscriptionValidator, { GetAllowEndpoints, SubscriptionEndpoint } from './subscriptionValidator';
+import validPatient from '../../../sampleData/validV4Patient.json';
+import invalidPatient from '../../../sampleData/invalidV4Patient.json';
+
+const search: Search = ElasticSearchService;
+
+const getSubscriptionResource = (endpoint: string) => {
+    return {
+        resourceType: 'Subscription',
+        status: 'requested',
+        reason: 'Monitor Patients for Organization 123',
+        criteria: 'Patient?managing-organization=Organization/123',
+        channel: {
+            type: 'rest-hook',
+            endpoint,
+            payload: 'application/fhir+json',
+            header: ['Authorization: Bearer secret-token-abc-123'],
+        },
+    };
+};
+
+const getBundleResource = (resourceToAdd: { resourceType: string }) => {
+    return {
+        resourceType: 'Bundle',
+        type: 'transaction',
+        entry: [
+            {
+                fullUrl: 'urn:uuid:some-fake-uuid',
+                resource: invalidPatient,
+                request: {
+                    method: 'POST',
+                    url: 'Patient',
+                },
+            },
+            {
+                fullUrl: 'urn:uuid:another-fake-uuid',
+                resource: resourceToAdd,
+                request: {
+                    method: 'POST',
+                    url: resourceToAdd.resourceType,
+                },
+            },
+            {
+                request: {
+                    method: 'DELETE',
+                    url: 'Patient/some-fake-id',
+                },
+            },
+        ],
+    };
+};
+
+describe('Multi-tenancy mold: Validating Subscriptions', () => {
```

Comment:
I think that there's a bit too much repetition of tests for multi and single tenancy. We should find a way to only write them once. 

Try using `test.each` or `describe.each` (https://jestjs.io/docs/api#testeachtablename-fn-timeout) There are some examples already in our codebase.